### PR TITLE
test(grid): ensure cypress tests are not dependent on pod component

### PR DESCRIPTION
**Pull request by Parsium**
### Proposed behaviour

- Amend `Grid` cypress locators and the affected tests - so tests are independent of `Pod`.
- Amend `Grid` cypress tests for readability and fix any errors.

### Current behaviour

- `Grid` uses locators that are dependent on the `Pod` component.

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

No functionality or visuals to test - ensure all cypress tests are passing and the tests are testing the behaviour they describe.

### CODE REVIEW

File: src/components/grid/grid.test.js [hunk 1]
```diff
@@ -451,31 +349,24 @@ context("Testing Grid component", () => {
...
         .and("have.css", "align-self", "stretch")
         .and("have.css", "justify-self", "stretch");
-      gridComponent()
-        .children()
-        .eq(1)
+      gridItem(1)
         .should("have.css", "grid-column", "6 / 13")
         .and("have.css", "grid-row", "1 / 1")
-        .and("have.css", "width", "8px")
+        .and("have.css", "width", "75.328125px")
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:366]: we should round this value :)

File: src/components/grid/grid.test.js [hunk 2]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
-            .then(parseFloat)
-            .then(($el) => {
-              expect($el).to.be.gte(podWidth);
-              expect($el).to.be.lt(podWidth + 1);
-            });
-        }
+        gridItem(itemNumber - 1).should("have.css", "grid-row", row);
+        gridItem(itemNumber - 1).should("have.css", "grid-column", col);
+        gridItem(itemNumber - 1)
+          .invoke("css", "width")
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:86]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 3]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:107]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 4]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
-            .invoke("css", "width")
-            .then(parseFloat)
-            .then(($el) => {
-              expect($el).to.be.gte(podWidth);
-              expect($el).to.be.lt(podWidth + 1);
-            });
-        }
+        gridItem(itemNumber - 1).should("have.css", "grid-row", row);
+        gridItem(itemNumber - 1).should("have.css", "grid-column", col);
+        gridItem(itemNumber - 1)
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:127]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 5]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
+          .invoke("css", "width")
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:149]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 6]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
-            .invoke("css", "width")
-            .then(parseFloat)
-            .then(($el) => {
-              expect($el).to.be.gte(podWidth);
-              expect($el).to.be.lt(podWidth + 1);
-            });
-        }
+        gridItem(itemNumber - 1).should("have.css", "grid-row", row);
+        gridItem(itemNumber - 1).should("have.css", "grid-column", col);
+        gridItem(itemNumber - 1)
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:169]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 7]
```diff
@@ -31,267 +32,167 @@ const viewportSize = (sizeOfViewport) => {
...
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:190]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(itemNumber - 1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 8]
```diff
@@ -451,36 +343,38 @@ context("Testing Grid component", () => {
...
         .should("have.css", "grid-column", "1 / 7")
         .and("have.css", "grid-row", "1 / 1")
-        .and("have.css", "width", "623px")
         .and("have.css", "align-self", "stretch")
         .and("have.css", "justify-self", "stretch");
-      gridComponent()
-        .children()
-        .eq(1)
+      gridItem(0)
+        .invoke("css", "width")
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:357]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(0).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 9]
```diff
@@ -451,36 +343,38 @@ context("Testing Grid component", () => {
...
         .should("have.css", "grid-column", "6 / 13")
         .and("have.css", "grid-row", "1 / 1")
-        .and("have.css", "width", "8px")
         .and("have.css", "align-self", "end")
         .and("have.css", "justify-self", "end");
-      gridComponent()
-        .children()
-        .eq(2)
+      gridItem(1)
+        .invoke("css", "width")
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:366]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(1).then(($el) => {
     expect($el.css("width"))...
})`

File: src/components/grid/grid.test.js [hunk 10]
```diff
@@ -451,36 +343,38 @@ context("Testing Grid component", () => {
...
+        .then(parseFloat)
+        .then((width) => expect(width).to.be.within(75, 76));
+      gridItem(2)
         .should("have.css", "grid-column", "1 / 13")
         .and("have.css", "grid-row", "3 / 3")
-        .and("have.css", "width", "1270px")
         .and("have.css", "align-self", "start")
         .and("have.css", "justify-self", "stretch");
+      gridItem(2)
```
Comment (DlgSHi) [inline src/components/grid/grid.test.js:374]: I would rather not use `invoke` here, just use the `jQuery` method using 

`gridItem(2).then(($el) => {
     expect($el.css("width"))...
})`

**Review by robinzigmond — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**